### ISSUE TITLE
dub run <package> should search the registry on failure

### ISSUE THREAD

**Issue by wilzbach** [dlang/dub#877]
Imho the following error message is a bit tricky to newcomers:

```
> dub run dlang-tour
Failed to find a package named 'dlang-tour'.
```

There are two easy solutions to this problem:

1) print a hint like 'Have you downloaded the package? Try `dub fetch <pkg>`
2) Automatically try to run `dub fetch` if `dub run` fails to find a package locally (preferred)

What do you prefer?

**Comment by andre2007** [comment]
https://github.com/dlang/dub/pull/1428

**Comment by Samathy** [comment]
This was addressed and merged in #1428 and should be closed as completed. @wilzbach
